src/model/implementation/time-to-live.ts

```diff
@@ -0,0 +1,131 @@
+import { GraphQLIncludeDirective, GraphQLInt } from 'graphql';
+import memorize from 'memorize-decorator';
+import { SCALAR_JSON } from '../../schema/constants';
+import { GraphQLDateTime } from '../../schema/scalars/date-time';
+import { TimeToLiveConfig, TypeKind } from '../config';
+import { ValidationMessage } from '../validation';
+import { ModelComponent, ValidationContext } from '../validation/validation-context';
+import { Field } from './field';
+import { Model } from './model';
+import { RootEntityType } from './root-entity-type';
+import { Type } from './type';
+
+export class TimeToLiveType implements ModelComponent {
+    constructor(readonly input: TimeToLiveConfig, readonly model: Model) {}
+
+    validate(context: ValidationContext): void {
+        this.traversePath(mess => context.addMessage(mess));
```

Comment:
we should validate `expireAfterDays` (positive integer) and `typeName` (exists and is a root entity type) here